sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -284,7 +293,7 @@ class FrameLessOffsetWindowFunctionFrame(
       }
   } else {
     (current: InternalRow) =>
-      if (inputIndex >= 0 && inputIndex < input.length) {
+      if (inputIndex >= 0 && input != null && inputIndex < input.length) {
```

Comment:
why this change?

Reply:
If `Math.abs(offset) > rows.length` is true, no need to assign a value to `input`.

Reply:
Why we don't need the null check before? where did we instantiate `input`?

Reply:
Because the origin code always call `resetStates`.

Reply:
I feel the code is a bit subtle now. In the branch above here, there is no null check: `while (nextSelectedRow == EmptyRow && inputIndex < input.length) {`